### PR TITLE
Bugfix FXIOS-6797 [v116] dismiss the alert that's preventing the screen to dismiss if the app goes to background (#15134)

### PR DESCRIPTION
**Pull request by petruSt24**
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-6797)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/15134)

### Description
dismissed the remove card alert if the app goes to background

### Pull requests checks where applicable
- [X] Fill in the three TODOs above (tickets number and description of your work)
- [X] The PR name follows our [naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Accessibility implemented and tested (minimum Dynamic Text and VoiceOver)
- [ ] Unit tests written and passing
- [ ] Documentation / comments for complex code and public methods

### CODE REVIEW

**Comment by OrlaM** [comment]
This makes sense to me but my swiftUI is not great so adding Nish to review just in case I'm missing something.

File: Client/Frontend/Autofill/CreditCard/ViewComponents/RemoveCardButton.swift [hunk 1]
```diff
@@ -62,6 +62,9 @@ struct RemoveCardButton: View {
         .onChange(of: themeVal) { newThemeValue in
             applyTheme(theme: newThemeValue.theme)
         }
+        .onReceive(NotificationCenter.default.publisher(for: UIApplication.willResignActiveNotification)) { _ in
+            showAlert = false
+        }
```
Comment (nbhasin2) [inline Client/Frontend/Autofill/CreditCard/ViewComponents/RemoveCardButton.swift:67]: You could try `@Environment (\.scenePhase) private var scenePhase` however using the publisher is fine as I have had reliability issues by using `scenePhase` environment variable. Maybe it's fixed now 🤷🏼

Comment (nbhasin2) [inline Client/Frontend/Autofill/CreditCard/ViewComponents/RemoveCardButton.swift:67]: ⚠️Nit: Indent
```suggestion 
        .onReceive(
            NotificationCenter.default.publisher(for: UIApplication.willResignActiveNotification)
        ) { _ in
            showAlert = false
        }
```

Comment (nbhasin2) [reply Client/Frontend/Autofill/CreditCard/ViewComponents/RemoveCardButton.swift:67]: @petruSt24, can you comment on why you are making `showAlart = false` as you need the alert to dismiss before dismissing the view?

A slight drawback with this approach is that even after a user authenticates, you still don't see the alert. Confirm with UX before merging this please

Comment (petruSt24) [reply Client/Frontend/Autofill/CreditCard/ViewComponents/RemoveCardButton.swift:67]: there's a mechanism in place that handles the dismissal of screens when authentication fails. That mechanism will fail when an alert controller is also present in that screen and the RemoveCardButton doesn't nor should it have access to the user's authentication status for security reasons

Comment (nbhasin2) [reply Client/Frontend/Autofill/CreditCard/ViewComponents/RemoveCardButton.swift:67]: @petruSt24 the comment goes inside the onReceive not on github 😅 
Summarize it in a sentence so folks are aware when they look at the code please

**Review by nbhasin2 — APPROVED** [review]

**Comment by lmarceau** [comment]
FYI Swiftlint failed on BR

**Comment by nbhasin2** [comment]
@petruSt24 can you fix the rebase issues, there seems to be a odd rebase here

**Pull request closed without merging**